pallets/otc-settlements/src/lib.rs

```diff
@@ -0,0 +1,542 @@
+// This file is part of galacticcouncil/warehouse.
+// Copyright (C) 2020-2023  Intergalactic, Limited (GIB). SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{pallet_prelude::*, PalletId};
+use frame_system::{
+	ensure_none,
+	offchain::{SendTransactionTypes, SubmitTransaction},
+	pallet_prelude::{BlockNumberFor, OriginFor},
+	RawOrigin,
+};
+use hydradx_traits::router::{
+	AmmTradeWeights, AmountInAndOut, AssetPair, RouteProvider, RouteSpotPriceProvider, RouterT, Trade,
+};
+use orml_traits::{GetByKey, MultiCurrency};
+use pallet_otc::weights::WeightInfo as OtcWeightInfo;
+pub use pallet_otc::OrderId;
+use sp_arithmetic::traits::{CheckedMul, Saturating};
+use sp_arithmetic::{ArithmeticError, FixedPointNumber, FixedU128};
+use sp_runtime::offchain::storage::StorageValueRef;
+use sp_runtime::offchain::storage_lock::{StorageLock, Time};
+use sp_runtime::offchain::Duration;
+use sp_runtime::traits::AccountIdConversion;
+use sp_std::vec;
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarks;
+
+pub mod weights;
+
+use weights::WeightInfo;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+pub type Balance = u128;
+pub type NamedReserveIdentifier = [u8; 8];
+
+pub const PALLET_ID: PalletId = PalletId(*b"otcsettl");
+
+pub const NAMED_RESERVE_ID: NamedReserveIdentifier = *b"otcorder";
+
+// value taken from https://github.com/substrate-developer-hub/recipes/blob/master/pallets/ocw-demo/src/lib.rs
+pub const UNSIGNED_TXS_PRIORITY: u64 = 100;
+
+pub const OFFCHAIN_WORKER_DATA: &[u8] = b"hydradx/otc-settlements/data/";
+pub const OFFCHAIN_WORKER_DATA_LAST_UPDATE: &[u8] = b"hydradx/otc-settlements/data-last-update/";
+pub const OFFCHAIN_WORKER_LOCK: &[u8] = b"hydradx/otc-settlements/lock/";
+pub const OFFCHAIN_WORKER_MAX_ITERATIONS: &[u8] = b"hydradx/otc-settlements/max-iterations/";
+pub const LOCK_DURATION: u64 = 5_000; // 5 seconds
```

Comment:
these constants needs to be more descriptive, especially when they are used only in one place

Reply:
something like SORTED_ORDERS_LOCK and FILL_SEARCH_ITERATIONS would be better